### ISSUE TITLE
[java] Drop AtLeastOneConstructor

### ISSUE THREAD

**Issue by mcandre** [pmd/pmd#1609]
**Affects PMD Version:** 

1.04+

**Rule:**

[AtLeastOneConstructor](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#atleastoneconstructor)

**Description:**

PMD complains when no explicit constructor is declared for a class. Including JUnit tests.

Note that JUnit expects tests to feature a public, 0-ary constructor, or else it throws an initialization error. So treating JUnit tests as utility classes with private constructors is not an option.

One could explicitly declare an empty constructor for each JUnit test, but that's wasteful. In any case, Java automatically creates an implicit constructor, so there's really no need for PMD to generate noise about this.

As a workaround, I am disabling the AtLeastOneConstructor rule entirely, in my rulesets.xml:

```xml
<rule ref="category/java/codestyle.xml">
    <exclude name="AtLeastOneConstructor" />
</rule>
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Gradle

**Comment by Fernal73** [comment]
This rule could be enhanced so that parameterless constructor body is never empty. You don't want an empty public constructor declared in this case when the compiler provides one. This is not the same as UncommentedEmptyConstructor. An empty constructor (parameterless) should simply not be allowed.

**Comment by jsotuyod** [comment]
This is a codestyle rule, and users are encouraged to define their own rulesets rather than blindly import whole categories.
